tests/core/test_filter_pred.py

```diff
@@ -0,0 +1,88 @@
+import asyncio
+
+import pytest
+
+from conftest import (
+    generate_unique_name,
+)
+from lahja import (
+    BaseEvent,
+    ConnectionConfig,
+    Endpoint,
+)
+
+
+@pytest.mark.asyncio
+async def test_subscribed_events(event_loop: asyncio.AbstractEventLoop) -> None:
+    class TestSubscriptionEvent(BaseEvent):
+        pass
+
+    endpoint = Endpoint()
+    endpoint.subscribe(TestSubscriptionEvent, lambda event: None)
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent}
+
+    class TestStreamEvent(BaseEvent):
+        pass
+
+    async def stream() -> None:
+        async for _ in endpoint.stream(TestStreamEvent):  # noqa: F841
+            pass
+
+    future = asyncio.ensure_future(stream())
+    await asyncio.sleep(0)  # give stream a chance to be scheduled
+
+    assert endpoint.subscribed_events == {TestSubscriptionEvent, TestStreamEvent}
+
+    future.cancel()
+    endpoint.stop()
+
+
+class TestEvent(BaseEvent):
```

Comment:
Annoying but calling this `EventForTesting` will help prevent pytest from collecting it as a test.